src/main/java/io/xcherry/core/persistence/Persistence.java

```diff
@@ -1,125 +1,121 @@
+                            encoder.decodeFromString(column.getQueryValue(), columnValueType)
+                        );
+                    } else {
+                        otherColumnMap.put(
+                            column.getColumn(),
+                            encoder.decodeFromString(column.getQueryValue(), columnValueType)
+                        );
+                    }
+                }
+
+                appDatabase
+                    .get(table.getTableName())
+                    .add(AppDatabaseRow.create(table.getTableName(), primaryKeyColumnMap, otherColumnMap));
+            }
+        }
     }
 
     /**
-     * Get a global attribute as the column value from the specified table column.
+     * Get rows from the specified app database table.
      *
-     * @param tableName     the table where the global attribute is stored.
-     * @param columnName    the column of the table where the global attribute is stored.
-     * @return  the column value as the global attribute.
+     * @param tableName table name.
+     * @return  rows of the table.
      */
-    public Object getGlobalAttribute(final String tableName, final String columnName) {
-        if (!globalAttributes.containsKey(tableName)) {
-            throw new GlobalAttributeNotFoundException(
-                String.format("Table %s does not exist within the global attributes", tableName)
-            );
-        }
-        if (!globalAttributes.get(tableName).containsKey(columnName)) {
-            throw new GlobalAttributeNotFoundException(
-                String.format(
-                    "Column %s does not exist in the table %s within the global attributes",
-                    columnName,
-                    tableName
-                )
-            );
+    public List<AppDatabaseRow> getAppDatabaseRows(final String tableName) {
+        if (!appDatabase.containsKey(tableName)) {
+            throw new AppDatabaseNotFoundException(String.format("Table %s does not exist in app database", tableName));
         }
 
-        return globalAttributes.get(tableName).get(columnName);
+        return appDatabase.get(tableName);
     }
 
-    /**
-     * Upsert a global attribute.
-     * If a global attribute already exists in the column, update its value. Otherwise, insert a new row with the column value as the global attribute.
-     *
-     * @param tableName     the table where the global attribute will be stored.
-     * @param columnName    the column of the table where the global attribute will be stored.
-     * @param columnValue   the column value to upsert as the global attribute.
-     */
-    public void upsertGlobalAttribute(final String tableName, final String columnName, final Object columnValue) {
-        if (!globalAttributesToUpdate.containsKey(tableName)) {
-            globalAttributesToUpdate.put(tableName, new HashMap<>());
+    public AppDatabaseWrite getAppDatabaseWrite(final DatabaseStringEncoder encoder) {
```

Comment:
the encoder/converter can be eliminated here to make it more friendly. 

I also think the current APIs in the doc is more clear:
```
rows = persistence.readAppDatabaseTable(TableName)


row = rows.get(0) 
int rowIndex = row.getIndex()
value = row.getColumn(ColumnName)

rows.foreach (
   row ->{ … }
)

row = persistence.readAppDatabaseTableFirstRow(TableName)

persistence.writeAppDatabaseTableFirstRow(TableName, ColumnName, value) 
persistence.writeAppDatabaseTable(TableName, ColumnName, value,  [rowIndex])

```

Reply:
Now, the API is like:
```
row = rows.get(0)
row.upsert(xxx)
```

so users don't need to remember the index of each row, or remember the api on persistence level. They can directy update for each row. I think this way is more convenient to use.